test/compress/exports.js

```diff
@@ -142,3 +142,38 @@ mangle_rename: {
         }
     }
 }
+
+hoist_exports: {
+    options = {
+        evaluate: true,
+        hoist_exports: true,
+        reduce_vars: true,
+        toplevel: true,
+        unused: true,
+    }
+    mangle = {
+        toplevel: true,
+    }
+    input: {
+        const a = 42;
+        export let bbb, { foo: ccc } = a;
+        export function fff(d, { [bbb]: e }) {
+            d(e, fff);
+        }
+        export default a;
+        export default async function g(x, ...{ [ccc]: y }) {
```

Comment:
An ES module can only have a single `export default`.
This is an error in most parsers like acorn: `Duplicate export 'default' (7:15)`.
I suppose GIGO can work here, but an explicit parse error may be better.

Reply:
There are other stuff like inside `{ ... }` or duplicated aliases etc. that would throw as well, which is why I've picked GIGO and call it a day.

Reply:
Just curious &minus; would `acorn` complain about the following?
```js
export default 42;
export { console as default };
```

Reply:
```
$ echo 'export default 42; export { console as default };' | acorn --module --ecma2020
Duplicate export 'default' (1:39)
```